src/FieldDataPluginFramework.Serialization.UnitTests/JsonConfigTests.cs

```diff
@@ -236,23 +238,67 @@ private DischargeActivity CreateDischargeActivity()
 
         private ChannelMeasurementBase CreateChannelMeasurement()
         {
-            NextIsAdcp = !NextIsAdcp;
+            var channelMeasurement = CreateNextChannelMeasurement();
 
-            return NextIsAdcp
-                ? CreateAdcpMeasurement()
-                : CreateManualGauging();
+            NextMeasurementType = (ChannelMeasurementType) ((1 + (int) NextMeasurementType) % ChannelMeasurementTypeCount);
```

Comment:
Since the framework itself doesn't expose any property that says what type of channel measurement it is, I had to hack a bit here in the test code to ensure that it attempts to serialize and deserialize at least one instance of every possible channel measurement type.

So I created this local enumeration type just for the unit tests, and iterated through all possible types. Can you tell I cut my coding teeth in C and just want to cast everything to other data types? :)